tensorflow_datasets/image/chexpert.py

```diff
@@ -0,0 +1,118 @@
+"""https://stanfordmlgroup.github.io/competitions/chexpert/"""
+
+import csv
+import os
+
+import tensorflow_datasets.public_api as tfds
+import tensorflow as tf
+
+
+_DESCRIPTION = '''\
+CheXpert is a large dataset of chest X-rays and competition for automated chest 
+x-ray interpretation, which features uncertainty labels and radiologist-labeled 
+reference standard evaluation sets. It consists of 224,316 chest radiographs 
+of 65,240 patients, where the chest radiographic examinations and the associated 
+radiology reports were retrospectively collected from Stanford Hospital. Each 
+report was labeled for the presence of 14 observations as positive, negative, 
+or uncertain. We decided on the 14 observations based on the prevalence in the 
+reports and clinical relevance.
+
+The CheXpert dataset must be downloaded separately after reading and agreeing 
+to a Research Use Agreement. To do so, please follow the instructions on the 
+website, https://stanfordmlgroup.github.io/competitions/chexpert/.
+'''
+
+_CITATION = '''\
+@article{DBLP:journals/corr/abs-1901-07031,
+  author    = {Jeremy Irvin and Pranav Rajpurkar and Michael Ko and Yifan Yu and Silviana Ciurea{-}Ilcus and Chris Chute and Henrik Marklund and Behzad Haghgoo and Robyn L. Ball and Katie Shpanskaya and Jayne Seekins and David A. Mong and Safwan S. Halabi and Jesse K. Sandberg and Ricky Jones and David B. Larson and Curtis P. Langlotz and Bhavik N. Patel and Matthew P. Lungren and Andrew Y. Ng},
+  title     = {CheXpert: {A} Large Chest Radiograph Dataset with Uncertainty Labels and Expert Comparison},
+  journal   = {CoRR},
+  volume    = {abs/1901.07031},
+  year      = {2019},
+  url       = {http://arxiv.org/abs/1901.07031},
+  archivePrefix = {arXiv},
+  eprint    = {1901.07031},
+  timestamp = {Fri, 01 Feb 2019 13:39:59 +0100},
+  biburl    = {https://dblp.org/rec/bib/journals/corr/abs-1901-07031},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+'''
+
+# Path to images and category labels in data dir
+_DATA_DIR = 'CheXpert-v1.0-small'
+_TRAIN_DIR = os.path.join(_DATA_DIR, 'train')
+_VALIDATION_DIR = os.path.join(_DATA_DIR, 'valid')
+_TRAIN_LABELS_FNAME = os.path.join(_DATA_DIR, 'train.csv')
+_VALIDATION_LABELS_FNAME = os.path.join(_DATA_DIR, 'valid.csv')
+
+# Labels per category
+_LABELS = ['', '-1.0', '0.0', '1.0']
```

Comment:
Are there more meaningful names you could put here?

Reply:
Currently, each image has labels for 14 separate categories, where each label can take any value from `_LABELS`. Thus, `_LABELS` is consistent with those found in the dataset. I was under the assumption that the `names` for a `ClassLabel` must be the ones present in the data. Is there a way to make them more clear?

Reply:
Hmm..yeah, we want to represent the data faithfully, but clarity too is a virtue.

What are the semantic meanings of these values? We could do, for example:

```
# Maps to 0=negative, 1=positive, 2=uncertain, 3=unknown
_LABELS = OrderedDict{
  "-1.0": "negative",
  "1.0": "positive",
  "0.0": "uncertain",
  "": "unknown",
}

# In info
ClassLabel(names=_LABELS.values())

# In generate
yield {
  ...
  "label": _LABELS[label]
}
```

Reply:
The meanings are blank for unmentioned, 0 for negative, -1 for uncertain, and 1 for positive. I think your suggestion is a good way to maintain clarity! I will add that in 👍

Reply:
Do you mean `_LABELS.keys()`? for `ClassLabel` since we want the true data labels?

Reply:
Ah sorry on the previous comment - I understand what you meant now.